### PR TITLE
allow customization of rentals sync logic

### PR DESCRIPTION
**Pull request by StoneFrog**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 ### master
 
+* [BREAKING CHANGE] Drop support for rails prior to 6.0
+* [BREAKING CHANGE] Drop support for ruby prior to 2.7
+* Allow to inject custom rentals synchronizer that is executed before RentalsController#index action
```
Comment (Azdaroth) [inline CHANGELOG.md:5]: might be a good idea to document it in Readme, to not look for this in the code

File: spec/controllers/admin/rentals_controller_spec.rb [hunk 2]
```diff
@@ -16,15 +16,38 @@
...
-    before do
-      expect(Rental).to receive(:synchronize).with(scope: account) do
-        # pretending to sync rentals :P
-        rental
+    context "with default synchronization" do
+      before do
+        expect(Rental).to receive(:synchronize).with(scope: account) do
+          # pretending to sync rentals :P
+          rental
+        end
```
Comment (JureCindro) [inline spec/controllers/admin/rentals_controller_spec.rb:24]:
```suggestion
        expect(Rental).to receive(:synchronize).with(scope: account).and_return(rental) # pretending to sync rentals :P
```
you probably don't need a block, also this is a non-blocking comment 🥁

File: lib/bookingsync_portal.rb [hunk 3]
```diff
@@ -60,6 +60,9 @@ module BookingsyncPortal
   # message bus channel scope
   mattr_accessor :message_bus_channel_scope
 
+  mattr_accessor :rentals_synchronizer
```
Comment (maclunar) [inline lib/bookingsync_portal.rb:63]: a line of explanation in a comment above this mattr_accessor definition would be welcome - to have some more clarity and also to keep it consistent with the rest of the file

**Review by JureCindro — APPROVED** [review]

**Review by maclunar — APPROVED** [review]

File: spec/controllers/admin/rentals_controller_spec.rb (hunk 2 above)
Comment (StoneFrog) [reply spec/controllers/admin/rentals_controller_spec.rb:24]: it seems to be evaluated before running get :index, so count doesn't change

Comment (JureCindro) [reply spec/controllers/admin/rentals_controller_spec.rb:24]: I see, yeah without a block it would be evaluated before.. nevermind then 😅

**Pull request merged**